networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -91,17 +94,40 @@ protected SafeFuture<?> doStart() {
     synchronized (this) {
       staticPeers.forEach(this::createPersistentConnection);
     }
-    periodicPeerSearch =
-        asyncRunner.runWithFixedDelay(
-            this::searchForPeers,
-            DISCOVERY_INTERVAL,
-            error -> LOG.error("Error while searching for peers", error));
-    connectToBestPeers(emptyList());
-    searchForPeers();
+    startSearchPeerTask();
     peerConnectedSubscriptionId = network.subscribeConnect(this::onPeerConnected);
     return SafeFuture.COMPLETE;
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private void startSearchPeerTask() {
+    this.periodicPeerSearch =
+        asyncRunner.runCancellableAfterDelay(
+            () -> searchForPeers().thenPeek(__ -> createFollowupSearchPeerTask()), Duration.ZERO);
+  }
+
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private void createFollowupSearchPeerTask() {
+    final int currentStage = startupCountdownStage.getAndDecrement();
+    if (currentStage < 0) {
+      return;
+    }
+    if (currentStage > 0) {
+      // Startup task, run for [startupCountdownStage] times
+      this.periodicPeerSearch =
+          asyncRunner.runCancellableAfterDelay(
+              () -> searchForPeers().thenPeek(__ -> createFollowupSearchPeerTask()),
```

Comment:
This probably should use `alwaysRun(...).finish` like above. In fact it seems to be exactly the same code so best to extract it as a new method and reuse it.